README.md

```diff
@@ -11,4 +11,241 @@ make setup
 Start pitaya admin
 ```
 make run
-```
\ No newline at end of file
+```
+
+# API
+
+**Errors**
+---
+Whenever pitaya admin respond with an error, it will be in the following format:
+`````
+{"success":false, "message":[msg], "reason": [error]}
+`````
+
+**Client Request**
+---
+Establishes a websocket connection with pitaya admin that can be used to send requests to handlers and receives its outputs through a pitaya client connected to given server address.
+
+* **URL**
+
+    /request
+
+* **Method**
+
+    `GET`
+
+* **URL Params**
+
+    **Required:**
+
+    `address=[pitaya server address]`
+
+* **Data Params**
+
+    After the websocket connection is set, pitaya admin will listen to requests in the following structure:
+
+    `````
+        Route       Pitaya route to handler
+        Payload     The data that will be sent to the handler
+        IsRequest   Whether this should be sent as a request or a notify
+    `````
+
+* **Response Example**
+
+    After the websocket connection is set and the pitaya client connects to the given server address, any message sent to the client will be forwarded as a response. It will return an error if it fails to establish any of the connections.
+
+
+**Documentation**
+---
+Returns server auto documentation given its type. Target server must implement pitaya's auto documentation remote.
+* **URL**
+
+    /docs
+
+* **Method**
+
+    `GET`
+
+* **URL Params**
+
+    **Optional:**
+
+    `methodtype=[either "remote" or "handler"]`
+
+    Specifies whether to get documentation only for remote's or handler's methods.
+
+    `route=[pitaya route]`
+
+    Specifies a single method route to get its documentation. Must have specified `methodtype` as well.
+
+    **Required:**
+
+    `type=[server type]`
+
+
+* **Response Example**
+
+   Auto documentation as implemented in remote or an error.
+
+* **Requirements**
+
+    As stated above, in order to documentation routes work, you will need to implement a remote for pitaya's auto documentation. A simple example is presented below.
```

Comment:
r/in order to documentation routes work/in order for documentation routes to work/